evm_arithmetization/src/cpu/control_flow.rs

```diff
@@ -78,6 +78,12 @@ pub(crate) fn eval_packed_generic<P: PackedField>(
     );
     yield_constr.constraint_transition(is_prover_input * (lv.is_kernel_mode - nv.is_kernel_mode));
 
+    // Check the helper value in the general columns.
+    yield_constr.constraint(
+        lv.op.push_prover_input
+            * (P::ONES - lv.is_kernel_mode - lv.general.push().push_prover_input_not_kernel),
```

Comment:
Nit: so that you can leverage `mul_sub` in the recursive version
```suggestion
        lv.op.push_prover_input
            * ((lv.is_kernel_mode + lv.general.push().push_prover_input_not_kernel) - P::ONES),
```